### DISCUSSION TITLE
Create Periodic Tasks

### DISCUSSION THREAD

**Discussion by alihatamitajik** [iddm/thread-priority#21] [Q&A]
Hi,
I want to create real-time tasks with ThreadPriority::Deadline: 
```rust
set_thread_priority_and_policy(thread_id,
                    ThreadPriority::Deadline { 
                        runtime: Duration::from_millis(200), 
                        deadline: Duration::from_millis(250), 
                        period: Duration::from_millis(1000) 
                    },
                    ThreadSchedulePolicy::Realtime(RealtimeThreadSchedulePolicy::RoundRobin)).unwrap();
```

and I got the following error:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Priority("Deadline is non-POSIX and cannot be converted.")'
```

I have two question:
1. Is it the right way to make periodic execution of the thread?
2. What should I do to make this Error disappear? 

Examples of using different thread policies and thread priorities would be nice!

I'm using Ubuntu 22.04.
@vityafx

**Comment by iddm** [comment]
It is really weird that I haven't noticed this message up until now! I must ask, is this something you need help with?

P.S. I'll try adding more examples for different thread policies and priorities as you suggested.

**Reply by iddm** [reply]
I took a look and couldn't find the problem, as the code has changed already. If this is still an issue please try to explain it once more.
